server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -525,4 +531,97 @@ public KnnVectorsFormat getKnnVectorsFormatForField() {
             return new Lucene94HnswVectorsFormat(hnswIndexOptions.m, hnswIndexOptions.efConstruction);
         }
     }
+
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        if (copyTo.copyToFields().isEmpty() != true) {
+            throw new IllegalArgumentException(
+                "field [" + name() + "] of type [" + typeName() + "] doesn't support synthetic source because it declares copy_to"
+            );
+        }
+        if (indexed) {
+            return new IndexedSyntheticFieldLoader();
+        }
+        return new DocValuesSyntheticFieldLoader();
+    }
+
+    private class IndexedSyntheticFieldLoader implements SourceLoader.SyntheticFieldLoader {
+        private VectorValues values;
```

Comment:
We've been using the lucene APIs directly for `SyntheticFieldLoader` subclasses. I believe it saves allocating a `float[]` on the doc values version which is nice, but not a huge thing. But I did it just to line up with the other implementations.